src/components/manager.ts

```diff
@@ -200,155 +211,139 @@ export class Manager {
         const rootFilesExcludePatterns = configuration.get('latex.search.rootFiles.exclude') as string[]
         const rootFilesExcludeGlob = rootFilesExcludePatterns.length > 0 ? '{' + rootFilesExcludePatterns.join(',') + '}' : undefined
         try {
-            const urls = await vscode.workspace.findFiles(rootFilesIncludeGlob, rootFilesExcludeGlob)
-            for (const url of urls) {
-                const content = utils.stripComments(fs.readFileSync(url.fsPath).toString(), '%')
+            const files = await vscode.workspace.findFiles(rootFilesIncludeGlob, rootFilesExcludeGlob)
+            for (const file of files) {
+                const content = utils.stripComments(fs.readFileSync(file.fsPath).toString(), '%')
                 const result = content.match(regex)
                 if (result) {
-                    const file = url.fsPath
-                    this.extension.logger.addLogMessage(`Try root file in root directory: ${file}`)
-                    const window = vscode.window
-                    if (window && window.activeTextEditor && this.isRoot(url.fsPath, window.activeTextEditor.document.fileName, true)) {
-                        this.extension.logger.addLogMessage(`Found root file in root directory: ${file}`)
-                        return file
-                    }
+                    this.extension.logger.addLogMessage(`Found root file from workspace: ${file.fsPath}`)
+                    return file.fsPath
                 }
             }
         } catch (e) {}
         return undefined
     }
 
-    isRoot(root: string, file: string, updateDependent = false): boolean {
-        if (!fs.existsSync(root)) {
-            return false
-        }
-        if (root === file) {
-            return true
-        }
-        if (updateDependent) {
-            this.findDependentFiles(root, undefined, true)
-            this.findAdditionalDependentFilesFromFls(root, true)
-        }
-        if (!this.texFileTree.hasOwnProperty(root) || !this.texFileTree.hasOwnProperty(file)) {
-            return false
-        }
-        for (const r of this.texFileTree[root]) {
-            if (this.isRoot(r, file)) {
-                return true
+    private getDirtyContent(file: string, reload: boolean = false) : string {
+        for (const cachedFile of Object.keys(this.cachedContent)) {
+            if (reload) {
+                break
             }
+            if (path.relative(cachedFile, file) !== '') {
+                continue
+            }
+            return this.cachedContent[cachedFile].content
         }
-        return false
+        const fileContent = utils.stripComments(fs.readFileSync(file, 'utf-8').toString(), '%')
```

Comment:
Why can we assume utf-8?

Reply:
I don't know 🤣 It was as is. Will remove it.